src/test/java/StringManipulationTest.java

```diff
@@ -19,130 +21,179 @@ public void tearDown() {
     }
 
-    @Test
+    @Test // Tests a string with two tokens
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("1 2");
+        int length = manipulatedstring.count();
+        assertEquals(2, length);
     }
 
-    @Test
+    @Test // Tests standard usage with every other character removed
     public void testRemoveNthCharacter1() {
-        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
+        manipulatedstring.setString("123456789");
+        assertEquals("13579", manipulatedstring.removeNthCharacter(2,false),manipulatedstring.getString());
     }
 
-    @Test
+    @Test // Tests standard usage with every other character replaced with a space
     public void testRemoveNthCharacter2() {
-        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        manipulatedstring.setString("123456789");
+        assertEquals("1 3 5 7 9", manipulatedstring.removeNthCharacter(2,true),manipulatedstring.getString());
     }
 
-    @Test
+    @Test // Tests removal from empty string
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("");
+        assertEquals("", manipulatedstring.removeNthCharacter(0,false),manipulatedstring.getString());
     }
 
-    @Test
+    @Test // Tests removing every 0th character
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("123456789");
+        assertEquals("123456789", manipulatedstring.removeNthCharacter(0,false),manipulatedstring.getString());
     }
 
-    @Test
+    @Test // Tests replacing every 0th character with a space
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("123456789");
+        assertEquals("123456789", manipulatedstring.removeNthCharacter(0,true),manipulatedstring.getString());
     }
 
-    @Test
+    @Test // Tests removing every character
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("123456789");
+        assertEquals("", manipulatedstring.removeNthCharacter(1,false),manipulatedstring.getString());
     }
 
-    @Test
+    @Test // Tests replacing every character with a space
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("123456789");
+        assertEquals("         ", manipulatedstring.removeNthCharacter(1,true),manipulatedstring.getString());
     }
 
-    @Test
-    public void testGeSubStrings1() {
-        manipulatedstring.setString("This is my string");
```

Comment:
random method in between

Reply:
The - before the line indicates that it was removed from the last commit